2022 theme stylings for product and shop pages.

vedanshujain opened the pull request:
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

This PR contains styling support for 2022 theme support for product and shop page.

Closes #31549.

### How to test the changes in this Pull Request:

1. Switch to WP 5.9 beta.
2. Switch to 2022 theme.
3. View shop and individual product page.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

<!-- Mark completed items with an [x] -->

### Changelog entry

> Enhancement -  Add support for 2022 theme shop and product pages.

### FOR PR REVIEWER ONLY:

* [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

vedanshujain committed:
2022 theme stylings for product and shop pages.

vedanshujain committed:
Add support for variable products and gallery.

vedanshujain committed:
Add margin to reset button.

vedanshujain committed:
Add support grouped product.

jeffstieler commented on plugins/woocommerce/includes/theme-support/class-wc-twenty-twenty-two.php:
```diff
@@ -0,0 +1,85 @@
+	public static function enqueue_admin_styles() {
+		wp_enqueue_style(
+			'woocommerce-twenty-twenty-one-admin',
+			str_replace( array( 'http:', 'https:' ), '', WC()->plugin_url() ) . '/assets/css/twenty-twenty-two-admin.css',
```
Should we create a stub source SCSS file for this so it exists?

vedanshujain replied:
I guess it's not needed? I will remove it.

jeffstieler reviewed (changes requested):
Most of this is looking good. I noticed a few differences compared to the Figma designs:

![Screen Shot 2022-01-04 at 10 08 10 AM](https://user-images.githubusercontent.com/63922/148098579-058551b6-7441-4be1-bc51-c720ada82396.png)

* Content width should be 1000px
* The breadcrumbs should not be rendered
* The horizontal divider should not be rendered (looks like this is in the notices? perhaps attach to the notices and not the wrapper)

![Screen Shot 2022-01-04 at 10 18 17 AM](https://user-images.githubusercontent.com/63922/148098604-a0142876-6632-47d4-9523-33f3649b91c0.png)

I think the font size for the "Shop" page title is a little larger than expected (though Figma shows that as an H3 which is incorrect..)

On the single product page:

![Screen Shot 2022-01-04 at 10 39 50 AM](https://user-images.githubusercontent.com/63922/148100863-14f1ab79-dbdc-456e-8819-81dbeab11a52.png)

* Max width should be 1000px
* Zoom icon in product image is different from designs
* Font sizes do not match designs

jeffstieler commented:
I've got a fix for the notices border in https://github.com/woocommerce/woocommerce/commit/43619bbf90977dd600a5c08405ef71a70ef8c55e.

vedanshujain committed:
Disable breadcrumb and add notice styling child.

vedanshujain committed:
Reduce heading size on title page.

vedanshujain committed:
Reduce font-size for content in tabs.

vedanshujain commented:
Thanks for the detailed review, much appreciated @jeffstieler 🙌 

> Content width should be 1000px

This is currently set by the theme to be 650px. Should we be overwriting this?

> The breadcrumbs should not be rendered

Addressed in https://github.com/woocommerce/woocommerce/pull/31536/commits/49b5a049efcc39f27da951e481951d66a61f2531.

> The horizontal divider should not be rendered (looks like this is in the notices? perhaps attach to the notices and not the wrapper)

Addressed in https://github.com/woocommerce/woocommerce/pull/31536/commits/49b5a049efcc39f27da951e481951d66a61f2531. Your fix works fine as well.

> I think the font size for the "Shop" page title is a little larger than expected 

Fixed in https://github.com/woocommerce/woocommerce/pull/31536/commits/6cd32f755bf0130f828137fedd13498fe0021bd7

> Zoom icon in product image is different from designs

I am bit struggling with this get this icon to show exactly as in the design. I would say current one is good enough, unless there is a neat way to do it which I am missing.

> Font sizes do not match designs

I reduced font sizes in tabs, but otherwise its not looking good in wide screens. Maybe we should have this verified again.

jeffstieler commented:
> This is currently set by the theme to be 650px. Should we be overwriting this?

Oh hm, all of the designs have 1000px as the width. I see that the `wideSize` is set to `1000px`...  we should ask the designer about this.

> I am bit struggling with this get this icon to show exactly as in the design. I would say current one is good enough, unless there is a neat way to do it which I am missing.

That works for me - I ended up having the same trouble with the icon in the select2 dropdowns.

vedanshujain commented:
> all of the designs have 1000px as the width. I see that the wideSize is set to 1000px

Ah you are right, looks like we need to add `.alignwide` class for the theme to set to 1000px. I will take care of this.

vedanshujain commented:
@jeffstieler Added another PR https://github.com/woocommerce/woocommerce/pull/31611 with remaining fixes, although that CSS code isn't pretty at all.

vedanshujain committed:
Fixes for PR 31536 (#31611)

* Set max width to 1000px for shop and products.

* Display breadcrumbs on product pages.

* Remove admin styles.

* Fix result count and ordering control alignment.

* Fix onsale alignment on single product pages.

Co-authored-by: Jeff Stieler <jeff.m.stieler@gmail.com>
